### ISSUE TITLE
Build instructions either outdated or incomplete

### ISSUE THREAD

**Issue by agc93** [Nexus-Mods/Vortex#7472]
After a previous attempt didn't go well, I just tried getting a Vortex build environment working locally and the `README` instructions are definitely either out-of-date or incomplete.

**Automatic**:
The bootstrap script is installing a *very* out-of-date and EOL'd version of Node and *only* installs the VS2015 build tools with no feedback on which options to choose (more on that below).

**Manual**:
I like to think I know Vortex and build systems pretty well at this point, and I simply **cannot** get a full Vortex build to work without some serious hacks. The vast majority of the problems come from node-gyp, but not all of them.

*ATL Headers*: Installing the build tools (either 2015 or 2017) with default options will not include all the components needed (that needs the C++ workload to be enabled) and also requires specifically enabling the "Visual C++ ATL for x86 and x64" component (_not_ the Spectre version, that breaks the build).

*Build Tools*: The README recommends installing the 2017 build tools but I was still having the build fail (even after setting `msvs_version`) until I *also* installed the 2015 equivalent. Also neither the 2015 or 2017 build tools are available at the link in the README (they've been moved to the [previous versions](https://visualstudio.microsoft.com/vs/older-downloads/) link)

*`gamebryo-savegame`*: The only way I could get a `yarn run build` to actually *complete* was to remove `gamebryo-savegame-management` from the subproject's JSON file, since it depends on `gamebryo-savegame` which appears to ignore the `msvs_version`. It actually works if you manually edit the `<source-root>\extensions\gamebryo-savegame-management\node_modules\gamebryo-savegame\package.json`'s `npm run build` to add the correct `--msvs_version` arg to the call to `node-gyp configure`, but the `BuildSubProjects.js` script will overwrite that change.

*Code Signing*: this one is a pretty minor one: `yarn dist` will default to enabling code signing which external devs definitely won't have set up, also failing the build.

---

Now I know that's basically a page of barely coherent ranting about problems with the worst part of JS builds that's mostly not in Vortex (aka `node-gyp`). That being said, if there's internal docs or something that are more complete/up-to-date it might be good to update the README with that since (without wanting to seem arrogant) if I can't get it working, I very much doubt many other potential contributors could get it working easily.

I don't even know what exactly the answer is: skipping extensions that fail to build, much more in-depth instructions, standardising all the extensions/dependencies on a single `msvs_version`, or something else entirely but I just wanted to raise this for tracking since this is the third time I've wanted to contribute a PR but have been unable to make a build work.

**Comment by TanninOne** [comment]
I've updated the bootstrap script and readme to use newer versions of the tools and specified the options that need to be selected (those have indeed changed)
The bootstrap script now produces a full working build on a completely fresh windows install again.

I'm not sure what your specific problem with gamebryo-savegame-management is though. It contains a native c++ module but it's not at all the only one and I've not run into any problem affecting this module only iirc.

**Comment by agc93** [comment]
Thanks @TanninOne I'll give it another shot and might have to spin out a separate VM so I can use the bootstrap script (already have most of those installed so don't want to duplicate).

As for `gamebryo-savegame`, it wasn't following the `msvs_version` and was always looking for a 4.0 ToolsVersion no matter what I did. Like I said, it built successfully by forcing `msvs_version` in the `package.json` but it wouldn't use the one from env vars or yarn config when I built. I'll try it again and raise a separate issue if I can reproduce it.
